pkg/operator/reporting/templates_test.go

```diff
@@ -0,0 +1,255 @@
+				ReportDataSources: []*metering.ReportDataSource{ds2},
+				PrestoTables: []*metering.PrestoTable{
+					NewPrestoTableCustom("", testNamespace, testNamespace, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					ReportingStart: reportStart,
+					ReportingEnd:   reportEnd,
+					Inputs:         map[string]interface{}{"UnreadyDeploymentReplicas": ds2.Name},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("error executing template: template: reportQueryTemplate:1:36: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: %s tableRef is empty", ds2.Name),
+			expectOutput: "",
+		},
+		{
+			name: "valid report query with valid templating but the presto table's table ref returns error and expected error message",
+			reportTemplate: &ReportQueryTemplateContext{
+				Namespace:         testNamespace,
+				Reports:           nil,
+				ReportQuery:       NewReportQueryCustom(ds1, "SELECT labels['namespace'], FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					NewPrestoTableCustom("", testNamespace, testNamespace, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					ReportingStart: reportStart,
+					ReportingEnd:   reportEnd,
+					Inputs:         map[string]interface{}{"UnreadyDeploymentReplicas": ds1.Name},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("error executing template: template: reportQueryTemplate:1:36: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: tableRef PrestoTable %s not found", ds1.Status.TableRef.Name),
+			expectOutput: "",
+		},
+		{
+			name: "ReportDataSource dependencies was not found (due to empty ds.Name) results in error and expected error message",
+			reportTemplate: &ReportQueryTemplateContext{
+				Namespace:         testNamespace,
+				Reports:           nil,
+				ReportQuery:       NewReportQueryCustom(ds1, "SELECT labels['namespace'], FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					NewPrestoTableCustom("", testNamespace, testNamespace, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					ReportingStart: reportStart,
+					ReportingEnd:   reportEnd,
+					Inputs:         map[string]interface{}{"UnreadyDeploymentReplicas": ""},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: "error executing template: template: reportQueryTemplate:1:36: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: ReportDataSource  dependency not found",
+			expectOutput: "",
```

Comment:
nit: You don't need to set `expectOutput` to "", that's the default, and since we're erroring, it won't be checked anyways

Reply:
Fixed.